sigstore/_internal/rekor/client.py

```diff
@@ -287,31 +289,20 @@ def __init__(self, url: str, pubkey: bytes, ctfe_pubkey: bytes) -> None:
             raise RekorClientError(f"Invalid public key type: {pubkey}")
         self._pubkey = pubkey
 
-        ctfe_pubkey = serialization.load_pem_public_key(ctfe_pubkey)
-        if not isinstance(
-            ctfe_pubkey,
-            (
-                rsa.RSAPublicKey,
-                ec.EllipticCurvePublicKey,
-            ),
-        ):
-            raise RekorClientError(f"Invalid CTFE public key type: {ctfe_pubkey}")
-        self._ctfe_pubkey = ctfe_pubkey
+        self._ct_keyring = ct_keyring
 
     def __del__(self) -> None:
         self.session.close()
 
     @classmethod
     def production(cls) -> RekorClient:
         return cls(
-            DEFAULT_REKOR_URL, _DEFAULT_REKOR_ROOT_PUBKEY, _DEFAULT_REKOR_CTFE_PUBKEY
+            DEFAULT_REKOR_URL, _DEFAULT_REKOR_ROOT_PUBKEY, CTKeyring.production()
```

Comment:
Hm, this would require larger changes than should be in this PR, but is there a separate client for CT? The CT log and Rekor are separate logs. The CT log is accessed at ctfe.sigstore.dev/<shard>, and while all Sigstore clients don't need to do online verification (since they only verify the SCTs offline), it's separate from rekor. I'd recommend splitting these apart, or at least associated CT with Fulcio rather than Rekor

Reply:
Yeah, @asraa brought this up as feedback here: https://github.com/sigstore/sigstore-python/issues/263#issuecomment-1286108466

I agree that this should be broken out, and I'll do that in a follow-up :slightly_smiling_face: